source/common/network/filter_manager_impl.cc

```diff
@@ -122,6 +122,7 @@ FilterStatus FilterManagerImpl::onWrite(ActiveWriteFilter* filter,
   // Filter could return status == FilterStatus::StopIteration immediately, close the connection and
   // use callback to call this function.
   if (connection_.state() != Connection::State::Open) {
+    buffer_source.getWriteBuffer().buffer.drain(buffer_source.getWriteBuffer().buffer.length());
```

Comment:
Thinking of why we didn't skip the flushout at
https://github.com/envoyproxy/envoy/blob/8d267ca634a21eb6b6922caa18dc69317ef4f65f/source/common/http/http1/codec_impl.cc#L264-L267

And `bytes_meter_` will be more accurate also.